--- a/drivers/net/wireless/ath/ath11k/core.c
+++ b/drivers/net/wireless/ath/ath11k/core.c
@@ -853,7 +853,7 @@ static const struct ath11k_hw_params ath
 		.svc_to_ce_map_len = 17,
 		.single_pdev_only = false,
 		.rxdma1_enable = true,
-		.num_rxmda_per_pdev = 1,
+		.num_rxdma_per_pdev = 1,
 		.rx_mac_buf_ring = false,
 		.vdev_start_delay = false,
 		.htt_peer_map_v2 = true,
--- a/drivers/net/wireless/ath/ath11k/pcic.c
+++ b/drivers/net/wireless/ath/ath11k/pcic.c
@@ -898,7 +898,7 @@ int ath11k_pcic_ext_config_gic_msi_irq(s
 {
 	u32 user_base_data = 0, base_vector = 0, base_idx;
 	struct ath11k_ext_irq_grp *irq_grp;
-	int j, ret = 0, num_vectors = 0;
+	int j, n, ret = 0, num_vectors = 0;
 	u32 num_irq = 0;
 
 	base_idx = ATH11K_PCI_IRQ_CE0_OFFSET + CE_COUNT_MAX;
@@ -910,8 +910,12 @@ int ath11k_pcic_ext_config_gic_msi_irq(s
 	irq_grp = &ab->ext_irq_grp[i];
 	irq_grp->ab = ab;
 	irq_grp->grp_id = i;
-	init_dummy_netdev(&irq_grp->napi_ndev);
-	netif_napi_add(&irq_grp->napi_ndev, &irq_grp->napi,
+	irq_grp->napi_ndev = alloc_netdev_dummy(0);
+	if (!irq_grp->napi_ndev)
+		return -ENOMEM;
+
+	init_dummy_netdev(irq_grp->napi_ndev);
+	netif_napi_add(irq_grp->napi_ndev, &irq_grp->napi,
 		       ath11k_pcic_ext_grp_napi_poll);
 
 	if (ab->hw_params.ring_mask->tx[i] ||
@@ -940,6 +944,10 @@ int ath11k_pcic_ext_config_gic_msi_irq(s
 		if (ret) {
 			ath11k_err(ab, "failed request irq %d: %d\n",
 				   irq_idx, ret);
+			for (n = 0; n <= j; n++) {
+				irq_grp = &ab->ext_irq_grp[n];
+				free_netdev(irq_grp->napi_ndev);
+			}
 			return ret;
 		}
 		ab->irq_num[irq_idx] = msi_desc->irq;
